wolfcrypt/src/asn.c

```diff
@@ -12400,7 +12408,15 @@ int GetFormattedTime(void* currTime, byte* buf, word32 len)
         hour = ts->tm_hour;
         mini = ts->tm_min;
         sec  = ts->tm_sec;
-        ret = XSNPRINTF((char*)buf, len,
+        #if defined(WOLF_C89)
+            if (len < 16) {
```

Comment:
Same. Avoid the hard coded 16. There must be a better way.

Reply:
see above.  the `16` comes from the `4` and the `2`s in the format string.  it's ad hoc literals all around.